### ISSUE TITLE
Unhandled 'error' event, like #1559, but from @node-redis/client, so internal...

### ISSUE THREAD

**Issue by ar4hc** [redis/node-redis#2026]
**Environment:**
 - **Node.js Version**: v17.6.0
 - **Redis Server Version**: Redis server v=6.2.6 sha=00000000:0 malloc=libc bits=64 build=c6f3693d1aced7d9
 - **Node Redis Version**: 4.0.4
 - **Platform**:  mac

i open a redis connection from l

```js
const client = createClient({
		url: envVars.REDIS_URL,
		retry_strategy: (options) => {
                         if (error?.code === 'ECONNREFUSED' || error?.code === 'NR_CLOSED' || error?.code === '??') {
				console.error(error.code);
				// take actions or throw exception
				console.error('The server refused the connection. Retrying connection...');
				return 5000;
			}
// [...]
```

and later the connection is made somewhat lazy:
```js
if (!queueClient.isOpen) {
			await queueClient.connect();
			console.info('connected to redis at', envVars.REDIS_URL);
		}
```
(BTW it's sveletekit app, and isOpen is in the hooks.js...)

It works if the app is started with or without a running redis server.
if redis is started after the app it reconnects fine (on the next try).
if i stop redis while the app is running,  the app crashes with this error:
 
```log
node:events:505
      throw er; // Unhandled 'error' event
      ^

SocketClosedUnexpectedlyError: Socket closed unexpectedly
    at Socket.<anonymous> ([...]/node_modules/.pnpm/@node-redis+client@1.0.4/node_modules/@node-redis/client/dist/lib/client/socket.js:182:118)
    at Object.onceWrapper (node:events:642:26)
    at Socket.emit (node:events:527:28)
    at TCP.<anonymous> (node:net:688:12)
Emitted 'error' event on Commander instance at:
    at RedisSocket.<anonymous> ([...]node_modules/.pnpm/@node-redis+client@1.0.4/node_modules/@node-redis/client/dist/lib/client/index.js:339:14)
    at RedisSocket.emit (node:events:527:28)
    at RedisSocket._RedisSocket_onSocketError (/[...]node_modules/.pnpm/@node-redis+client@1.0.4/node_modules/@node-redis/client/dist/lib/client/socket.js:205:10)
    at Socket.<anonymous> (/[...]node_modules/.pnpm/@node-redis+client@1.0.4/node_modules/@node-redis/client/dist/lib/client/socket.js:182:107)
    at Object.onceWrapper (node:events:642:26)
    at Socket.emit (node:events:527:28)
    at TCP.<anonymous> (node:net:688:12)
```

I have an on error handler, see above, still the error states there is none. 
As #1559  there could be an other client without an error handler, 
but redis is only used in one place in may code ... 

where should that missing handler be?

**Comment by Didas-git** [comment]
The missing handler would be on your first file.
You are a missing something like this:
```ts
client.on(err, () => { ... })
```

**Comment by ar4hc** [comment]
Oh, forgot to mention that...

i got a simple one for debugging:;

```js
client.on('error', (err) => {
		console.log('ⓘ on error:', err);
}
```
but that changed nothing, the handler isn't even called before the crash, so i assumed it wasn't necessary...

i had a whole battery of those:
```js
	// client.on('connect', log('redis connect'));
	// client.on('ready', log('redis ready'));
	// client.on('reconnecting', log('redis reconnecting'));
	// client.on('error', log('redis error'));
	// client.on('end', log('redis end'));
	/*
	client.on('reconnecting', () => {
		console.error('ⓘ redis reconnecting.');
	});
	*/
```
but some messed with `retry_strategy`, so they are not used anymore.
(i'm not sure which ones are active ...)

**Comment by 759325100** [comment]
retry_strategy remove to options->socket

**Comment by ar4hc** [comment]
> retry_strategy remove to options->socket

Sorry, i don't understand... 
"remove  from" options? socket? where...?

**Comment by leibale** [comment]
@ar4hc are you using `executeIsolated`/`isolated`? #2044

**Comment by ar4hc** [comment]
@leibale  Not that I'm aware of...

The complete code to initialize the connection looks like

```js
	const client = createClient({
		url: envVars.REDIS_URL,
		lazyConnect: true,
		showFriendlyErrorStack: true,
		retry_strategy: (options) => {
			const { error, total_retry_time, attempt } = options;
			if (error?.code === 'ECONNREFUSED' || error?.code === 'NR_CLOSED') {
				return 5000;
			}
			if (total_retry_time > 1000 * 15) {
				return undefined;
			}
			if (attempt > 10) {
				return undefined;
			}
			return Math.min(options.attempt * 1000, 5000); //in ms
		},
	});
```
and later

```js
if (!queueClient.isOpen) {
			await queueClient.connect();
}
// and then...
const blPopPromise = locals.queueClient.blPop(commandOptions({ isolated: true }), redisQueueName, redisTimeOut);
```

could it be implicit in there somewhere?

**Comment by leibale** [comment]
You are using the `commandOptions({ isolated: true })`, this bug is fixed in version 4.0.6.

Duplicate of #2044.
